staker/block_validator.go

```diff
@@ -1206,3 +1206,9 @@ func (v *BlockValidator) WaitForPos(t *testing.T, ctx context.Context, pos arbut
 		}
 	}
 }
+
+func (v *BlockValidator) GetLastValidated() validator.GoGlobalState {
+	v.reorgMutex.Lock()
```

Comment:
Since you only care about MessageIndex and not the full globalstate - easiest option would be to read it with holding reorg-read and calling validated()

Reply:
Thanks, done.